### DISCUSSION TITLE
How to address bundle size?

### DISCUSSION THREAD

**Discussion by benhodgson87** [nibtime/next-safe-middleware#72] [Q&A]
After installing this package, which has solved pretty much all our CSP headaches, we noticed that our built bundle sizes skyrocketed; 

![Screenshot 2022-08-10 at 16 56 25](https://user-images.githubusercontent.com/189707/183960003-d7248fea-27c0-41e9-89d9-a2b444e8625b.png)

Is there anything that can be done here to reduce the weight of this module?

**Comment by nibtime** [comment]
Hi @benhodgson87 

oh, that's problematic. Usually, the actual bundle size of the package wouldn't be a problem, as everything is in SSR only.

I think I know why that might be: The import of the wrapper of pages with `getServerSideProps`. Currently, it sits in the same bundle with the `_document` stuff, but it gets imported in page components. And since you need to compile to CommonJS for `_document`, I think the tree shaking doesn't work there and it imports the whole package.

So thank you very much for reporting this, I consider it a bug.

I think it is not too hard to resolve though, I just need to add the wrappers in a separate bundle that compiles to esm.
